HLTrigger/special/src/HLTMultipletFilter.cc

```diff
@@ -0,0 +1,162 @@
+  }
+  if (hltEtSumSeedLabel_ == edm::InputTag()) {
+    flag_[EtSum]    = false;
+  } else {
+    flag_[EtSum]    = true;
+    hltEtSumToken_  = consumes<l1t::EtSumBxCollection>(hltEtSumSeedLabel_);
+  }
+  if (hltJetSeedLabel_ == edm::InputTag()) {
+    flag_[Jet]      = false;
+  } else {
+    flag_[Jet]      = true;
+    hltJetToken_    = consumes<l1t::JetBxCollection>(hltJetSeedLabel_);
+  }
+  if (hltMuonSeedLabel_ == edm::InputTag()) {
+    flag_[Muon]     = false;
+  } else {
+    flag_[Muon]     = true;
+    hltMuonToken_   = consumes<l1t::MuonBxCollection>(hltMuonSeedLabel_);
+  }
+  if (hltTauSeedLabel_ == edm::InputTag()) {
+    flag_[Tau]      = false;
+  } else {
+    flag_[Tau]      = true;
+    hltTauToken_ = consumes<l1t::TauBxCollection>(hltTauSeedLabel_);
+  }
+  edm::LogVerbatim("Report") << "Input Parameters:: minN = " << minN_
+			     << " Bx Range = " << ibxMin_ << ":" << ibxMax_
+			     << " minPt = " << minPt_ << " Eta " << minEta_
+			     << ":" << maxEta_ << " Phi " << minPhi_ << ":" 
+			     << maxPhi_ << " GT Seed for EGamma " 
+			     << hltEGammaSeedLabel_ << ", EtSum "
+			     << hltEtSumSeedLabel_ << ", Jet "
+			     << hltJetSeedLabel_ << ", Muon "
+			     << hltMuonSeedLabel_ << ", and Tau "
+			     << hltTauSeedLabel_ << std::endl;
+}
+
+HLTMultipletFilter::~HLTMultipletFilter()= default;
+
+void
+HLTMultipletFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("L1EGammaInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1EtSumInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1JetInputTag",edm::InputTag("hltCaloStage2Digis:Jet"));
+  desc.add<edm::InputTag>("L1MuonInputTag",edm::InputTag());
+  desc.add<edm::InputTag>("L1TauInputTag",edm::InputTag("hltCaloStage2Digis:Tau"));
+  desc.add<int>("MinN",1);
+  desc.add<int>("IBxMin",0);
+  desc.add<int>("IBxMax",0);
+  desc.add<double>("MinEta",1.305);
+  desc.add<double>("MaxEta",3.000);
+  desc.add<double>("MinPhi",5.4105);
+  desc.add<double>("MaxPhi",5.5796);
+  desc.add<double>("MinPt", 20.0);
+  descriptions.add("hltMultipletFilter",desc);
+}
+
+bool HLTMultipletFilter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSetup, trigger::TriggerFilterObjectWithRefs & filterproduct) const {
+
+  // the filter object
+  if (saveTags()) {
+    if (flag_[EGamma]) filterproduct.addCollectionTag(hltEGammaSeedLabel_);
+    if (flag_[EtSum])  filterproduct.addCollectionTag(hltEtSumSeedLabel_);
+    if (flag_[Jet])    filterproduct.addCollectionTag(hltJetSeedLabel_);
+    if (flag_[Muon])   filterproduct.addCollectionTag(hltMuonSeedLabel_);
+    if (flag_[Tau])    filterproduct.addCollectionTag(hltTauSeedLabel_);
+  }
+
+  bool      accept(false);
+  int       nobj(0);
+
+  if ((!accept) && flag_[EGamma]) {
+    nobj += objects(iEvent,hltEGammaToken_,hltEGammaSeedLabel_,EGamma);
+    if (nobj >= minN_) accept = true;
```

Comment:
hi @bsunanda - it looks like this can be simplified to 

If ( nobj >= minN_ ) return true;

(and then you don't need the accept variable at all)